libp2p/switch.nim

```diff
@@ -284,7 +284,13 @@ proc newSwitch*(peerInfo: PeerInfo,
     connManager: connManager,
     peerStore: PeerStore.new(),
     dialer: Dialer.new(peerInfo.peerId, connManager, transports, ms, nameResolver),
-    nameResolver: nameResolver)
+    nameResolver: nameResolver,
+    listenError: listenError)
+
+  if switch.listenError.isNil:
+    switch.listenError = proc(e: ref CatchableError): ref CatchableError =
+      error "Failed to start one transport", error = e.msg
+      return nil
```

Comment:
I would actually fail by default, to keep the same behaviour as right now

Reply:
Updated.